src/storage/src/statistics.rs

```diff
@@ -859,30 +995,18 @@ impl AggregatedStatistics {
 
     /// Emit a snapshot of this workers local data.
     pub fn emit_local(
-        &self,
+        &mut self,
     ) -> (
-        Vec<(usize, SourceStatisticsUpdate)>,
-        Vec<(usize, SinkStatisticsUpdate)>,
+        Vec<(usize, SourceStatisticsRecord)>,
+        Vec<(usize, SinkStatisticsRecord)>,
     ) {
         // As an optimization, worker 0 does not broadcast it data. It ingests
         // it in `snapshot`.
         if self.worker_id == 0 {
             return (Vec::new(), Vec::new());
         }
 
-        let sources = self
-            .local_source_statistics
-            .values()
-            .flat_map(|(epoch, s)| s.snapshot().map(|v| (*epoch, v)))
-            .collect();
-
-        let sinks = self
-            .local_sink_statistics
-            .values()
-            .flat_map(|(epoch, s)| s.snapshot().map(|v| (*epoch, v)))
-            .collect();
-
-        (sources, sinks)
+        self._emit_local()
```

Comment:
Is this `_emit_local` on purpose? It looks like a non-standard variation of the `foo_inner` pattern or a leftover from development.

Reply:
it is correct! my previous pr meant to do this, but instead copied code!